src/epubcfi.js

```diff
@@ -152,15 +141,10 @@ EPUBJS.EpubCFI.prototype.parse = function(cfiStr) {
   path = pathComponent.split('/');
   end = path.pop();
 
-  cfi.steps = [];
-
-  path.forEach(function(part){
-    var step;
-
-    if(part) {
-      step = parseStep(part);
-      cfi.steps.push(step);
-    }
+  cfi.steps = path.filter(function(part) {
```

Comment:
Does this make it loop through all the parts twice? I'm all for replacing the `forEach`s with filter / map but not sure this instance makes sense.